spring-integration-mail/src/main/java/org/springframework/integration/mail/MailSendingMessageHandler.java

```diff
@@ -73,12 +74,15 @@ public String getComponentType() {
 
 	@Override
 	protected final void handleMessageInternal(Message<?> message) {
-		MailMessage mailMessage = this.convertMessageToMailMessage(message);
+		MailMessage mailMessage = convertMessageToMailMessage(message);
 		if (mailMessage instanceof SimpleMailMessage) {
 			this.mailSender.send((SimpleMailMessage) mailMessage);
 		}
 		else if (mailMessage instanceof MimeMailMessage) {
-			this.mailSender.send(((MimeMailMessage) mailMessage).getMimeMessage());
+			Assert.state(this.mailSender instanceof JavaMailSender,
+					"'mailSender' [" + this.mailSender + "] doesn't support 'MimeMailMessage' to send: " + mailMessage);
```

Comment:
We don't actually know if that is true; there might be some other `MailSender` implementation that __does__ support `MimeMailMessage`, but of course, we can't tell that (and don't know which method to call).

Maybe just change the message to "this adapter requires a JavaMailSender to send a MimeMailMessage" ?